awxkit/awxkit/api/pages/role_assignments.py

```diff
@@ -0,0 +1,48 @@
+import logging
+
+# from awxkit.api.mixins import DSAdapter, HasCreate, HasCopy
+# from awxkit.api.pages import (
+#     Credential,
+#     Organization,
+# )
+from awxkit.api.resources import resources
+
+# from awxkit.utils import random_title, PseudoNamespace, filter_by_class
+
+from . import base
+from . import page
+
+
+log = logging.getLogger(__name__)
+
+
+class RoleTeamAssignment(base.Base):
+    NATURAL_KEY = ('team', 'content_object', 'role_definition')
+
+
+page.register_page(
+    [resources.role_team_assignment, (resources.role_definition_team_assignments, 'post'), (resources.role_team_assignments, 'post')], RoleTeamAssignment
+)
+
+
+class RoleUserAssignment(base.Base):
+    NATURAL_KEY = ('user', 'content_object', 'role_definition')
+
+
+page.register_page(
+    [resources.role_user_assignment, (resources.role_definition_user_assignments, 'post'), (resources.role_user_assignments, 'post')], RoleUserAssignment
```

Comment:
The other thing worth noting is that assignments are immutable, meaning the detail endpoint, `resources.role_user_assignment`, would not allow PATCH/PUT... but it seems that no references to the detail endpoint exist here in the first place, so I guess we're good.